mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -0,0 +1,498 @@
+        ncFile = netCDF4.Dataset(outputFileClimo, mode='w')
+        # create dimensions
+        ncFile.createDimension('nz', len(refTopDepth))
+        for region in regionNames:
+            latBins = mocDictClimo['lat{}'.format(region)]['data']
+            mocTop = mocDictClimo['moc{}'.format(region)]['data']
+            ncFile.createDimension('nx{}'.format(region), len(latBins))
+        # create variables
+            x = ncFile.createVariable('lat{}'.format(region), 'f4',
+                                      ('nx{}'.format(region),))
+            x.description = 'latitude bins for MOC {}'\
+                            ' streamfunction'.format(region)
+            x.units = 'degrees (-90 to 90)'
+            y = ncFile.createVariable('moc{}'.format(region), 'f4',
+                                      ('nz', 'nx{}'.format(region)))
+            y.description = 'MOC {} streamfunction, annual'\
+                            ' climatology'.format(region)
+            y.units = 'Sv (10^6 m^3/s)'
+        # save variables
+            x[:] = latBins
+            y[:, :] = mocTop
+        depth = ncFile.createVariable('depth', 'f4', ('nz',))
+        depth.description = 'depth'
+        depth.units = 'meters'
+        depth[:] = refTopDepth
+        ncFile.close()
+    else:
+        # Read from file
+        print '   Read previously computed MOC streamfunction from file...'
+        ncFile = netCDF4.Dataset(outputFileClimo, mode='r')
+        refTopDepth = ncFile.variables['depth'][:]
+        mocDictClimo = {'depth': {'dims': ('nz'), 'data': refTopDepth}}
+        for region in regionNames:
+            latBins = ncFile.variables['lat{}'.format(region)][:]
+            mocTop = ncFile.variables['moc{}'.format(region)][:, :]
+            mocDictClimo['lat{}'.format(region)] = {
+                'dims': ('nx{}'.format(region)), 'data': latBins}
+            mocDictClimo['moc{}'.format(region)] = {
+                'dims': ('nz', 'nx{}'.format(region)), 'data': mocTop}
+        ncFile.close()
+
+    # Compute and plot time series of Atlantic MOC at 26.5N (RAPID array)
+    print '\n  Compute and/or plot post-processed Atlantic MOC '\
+          'time series...'
+    print '   Load data...'
+    ds = open_multifile_dataset(fileNames=dictTseries['inputFilesTseries'],
+                                calendar=calendar,
+                                simulationStartTime=simulationStartTime,
+                                timeVariableName='Time',
+                                variableList=variableList,
+                                variableMap=variableMap,
+                                startDate=dictTseries['startDateTseries'],
+                                endDate=dictTseries['endDateTseries'])
+    latAtlantic = mocDictClimo['latAtlantic']['data']
+    dLat = latAtlantic - 26.5
+    indlat26 = np.where(dLat == np.amin(np.abs(dLat)))
+    nyears = (dictTseries['endYearTseries'] -
+              dictTseries['startYearTseries'] + 1)
```

Comment:
Maybe you've addressed this already and I missed it but I think `nyears` needs to be the number of *complete* years (where you have data from all 12 months) for this to work.  There are a couple of issues here.  

First, `startYearTseries` and `endYearTseries` come from config options, not from the limits of the data set itself, so if the user puts and end year that's beyond the end of the data set (which we actually encourage if you want to plot the full time series), your loop over nyears will go beyond the end of the data.

Second, if the last year in the data set is incomplete, you want to exclude it because otherwise your loop over the 12 months below will go out of bounds.  If you allow a loop over less than 12 months during the last year, you would write out an incomplete data file for that year and that data file would not be completed the next time you run the analysis, even if the run has progressed and the year is now complete.

So I think you need to use:

```
startYear = days_to_datetime(ds.Time.min()).year
endDate = days_to_datetime(ds.Time.max())
endYear = endDate.year
if endDate.month < 12:
    endYear -= 1
nyears = (endYear - startYear + 1)
```